bundles/org.smarthomej.binding.tuya/README.md

```diff
@@ -117,6 +117,63 @@ The `min` and `max` parameters define the range allowed (e.g. 0-86400 for turn-o
 The `string` channel has one additional (optional) parameter `range`.
 It contains a comma-separated list of command options for this channel (e.g. `white,colour,scene,music` for the "workMode" channel).
 
+### Type `ir-code`
+
+IR code types:
++ `Tuya DIY-mode` - use study codes from real remotes. 
+   
+   Make a virtual remote control in DIY, learn virtual buttons.
+
++ `Tuya Codes Library (check Advanced options)` - use codes from templates library. 
+   
+   Make a virtual remote control from pre-defined type of devices.
+
+   Select Advanced checkbox to configure other parameters: 
+   + `irCode` - Decoding parameter
+   + `irSendDelay` - used as `Send delay` parameter
+   + `irCodeType` - used as `type library` parameter
+
++ `NEC` - IR Code in NEC format
++ `Samsung` - IR Code in Samsung format.
+
+**Additional options:**
+* `Active Listening` - Device will be always in learning mode. After send command with key code device stays in the learning mode
+* `DP Study Key` - **Advanced**. DP number for study key. Uses for receive key code in learning mode. Change it own your risk.
+
+
+If linked item received a command with `Key Code` (Code Library Parameter) then device sends appropriate key code.
+
+#### How to use IR Code in NEC format.
+
+Example, from Tasmota you need to use **_Data_** parameter, it can be with or without **_0x_**
+```json
+{"Time": "2023-07-05T18:17:42", "IrReceived": {"Protocol": "NEC", "Bits": 32, "Data": "0x10EFD02F"}}
+```
+
+Another example, use **_hex_** parameter
+```json
+{ "type": "nec", "uint32": 284151855, "address": 8, "data": 11, "hex": "10EFD02F" }
+```
+
+#### How to get key codes without Tasmota and other
+
+Channel can receive learning key (autodetect format and put autodetected code in channel).
+
+To start learning codes add new channel with Type String and DP = 1 and Range with `send_ir,study,study_exit,study_key`.
```

Comment:
Isn't that automatically added by discovery?

Reply:
may be
as i know, there are two types of ir controllers - old version (works via dp201) and new version (works via dp1,2,3,4,7,10,13)
How automaticaly detects it, I don't know

and new version  can work in two modes - DIY (now I can convert base64 tuya key code to nec format) and Tuya Library Mode, where key code consists from head and other data